jupyter_server/services/kernels/kernelmanager.py

```diff
@@ -474,4 +471,37 @@ def cull_kernel_if_idle(self, kernel_id):
                 idle_duration = int(dt_idle.total_seconds())
                 self.log.warning("Culling '%s' kernel '%s' (%s) with %d connections due to %s seconds of inactivity.",
                                  kernel.execution_state, kernel.kernel_name, kernel_id, connections, idle_duration)
-                self.shutdown_kernel(kernel_id)
+                await self.shutdown_kernel(kernel_id)
+
+
+# AsyncMappingKernelManager inherits as much as possible from MappingKernelManager,
+# overriding only what is different.
+class AsyncMappingKernelManager(MappingKernelManager, AsyncMultiKernelManager):
+
+    @default('kernel_manager_class')
+    def _default_kernel_manager_class(self):
+        return "jupyter_client.ioloop.AsyncIOLoopKernelManager"
+
+    def __init__(self, **kwargs):
+        self.pinned_superclass = AsyncMultiKernelManager
+        self.pinned_superclass.__init__(self, **kwargs)
+        self.last_kernel_activity = utcnow()
+        self.log.info("Async kernel manager started")
```

Comment:
Since we're now logging in serverapp that async kernel management is in use this is redundant and doesn't display the actual class name that is in use.

Reply:
Thanks @kevin-bates, I had not seen you already replied. Should we get rid of `self.last_kernel_activity = utcnow()` and other redundant logging?

Reply:
> Should we get rid of `self.last_kernel_activity = utcnow()`

I don't think so but I'm not sure why the timestamps differ.

Reply:
`self.last_kernel_activity = utcnow()` is not redundant.  Its necessary because we use the pinned_supclass's __init__() and `AsyncMultiKernelManager` doesn't know about `last_kernel_activity`.

I'm not sure what other logging is redundant.  I was just referencing the logging that indicates that the async kernel manager option has been "invoked".